### PR TITLE
Changed error message to reflect form field.

### PR DESCRIPTION
**Pull request by harshalbhatia**
The form asks date in DD-MM-YYYY format and error asked for YYYY-MM-DD format. Just a minor thing i noticed.

### CODE REVIEW

**Comment by jace** [comment]
What browser did you test in? DateField renders with `type='date'`, which some browsers recognise and render a custom widget for, using the browser's locale, but will submit back in `YYYY-MM-DD` format. Browsers that don't recognise it (like Safari?) render as plain text.

The error message hasn't been updated to recognise that some browsers do custom rendering these days. However, simply asking for DD-MM-YYYY format is incorrect.

**Comment by harshalbhatia** [comment]
I tested on both Desktop and Mobile chrome.
I was thinking the same thing actually. We can word the error message in a generic way instead of mentioning a date format.

**Comment by jace** [comment]
Since this is a `DataRequired` validator, let's make it `__("This is required")`

**Comment by jace** [comment]
FWIW, http://caniuse.com/#feat=input-datetime

Can we possibly use a polyfill to add support to broken browsers?

**Comment by harshalbhatia** [comment]
In issue #177 Balaji reports that the date picker doesn't work and that he got an error. I tried recreating the same (Safari, iPad, iOS 9) but the date picker worked fine. (Screenshots attached)
Might not be the same on Desktop Safari. I don't have access to a mac, I can't test that.

img_0159
img_0160

**Comment by jace** [comment]
Safari for iOS supports `input="date"` while Safari for macOS doesn't. [Go figure](http://caniuse.com/#feat=input-datetime). I'm guessing it has something to do with disagreements over [Shadow DOM](http://caniuse.com/#feat=shadowdom), which is the technical basis on which desktop browsers implement the widgets. Mobile browsers just use device-native widgets.

File: funnel/forms/space.py [hunk 1]
```diff
@@ -21,9 +21,9 @@ class ProposalSpaceForm(forms.Form):
     title = forms.StringField(__("Title"), validators=[forms.validators.DataRequired()])
     datelocation = forms.StringField(__("Date and Location"), validators=[forms.validators.DataRequired(), forms.validators.Length(max=50)])
     date = forms.DateField(__("Start date (for sorting)"),
-        validators=[forms.validators.DataRequired(__("Enter a valid date in YYYY-MM-DD format"))])
+        validators=[forms.validators.DataRequired(__("This field is required"))])
```
Comment (jace) [inline funnel/forms/space.py]: Let's use the same text in both places: "This is required"

**Comment by harshalbhatia** [comment]
Added a new commit after changing to a generic error message. Looking further into a viable polyfill.

**Pull request merged**